File: in_Python/0739 Daily Temperatures.py
class Solution(object):
    def dailyTemperatures(self, T):
        """
        :type T: List[int]
        :rtype: List[int]
        """
        st = []
        res = []
        for i in range(len(T)): res.append(0)
        for i in range(len(T)):
            while len(st) != 0 and T[i] > T[st[-1]]:
                res[st[-1]] = i - st[-1]
                st.pop()
            st.append(i)
        while len(st) != 0:
            res[st[-1]] = 0
            st.pop()
        return res
        

File: in_Java/0739 Daily Temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] T) {
        Stack<Integer> st = new Stack<>();
        int[] res = new int[T.length];
        for(int i = 0; i < T.length; i++){
            while(!st.isEmpty() && T[i] > T[st.peek()]){
                int idx = st.peek();
                res[idx] = i - idx;
                st.pop();
            }
            st.push(i);
        }
        return res;
    }
}